controllers/admin/AdminAjaxPs_buybuttonliteController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+* 2007-2016 PrestaShop
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+* @author    PrestaShop SA <contact@prestashop.com>
+* @copyright 2007-2015 PrestaShop SA
+* @license   http://addons.prestashop.com/en/content/12-terms-and-conditions-of-use
+* International Registered Trademark & Property of PrestaShop SA
+*/
+
+class AdminAjaxPs_buybuttonliteController extends ModuleAdminController
+{
+    public function ajaxProcessSearchProducts()
+    {
+        $context = Context::getContext();
+        $id_lang = $context->language->id;
+        $id_shop = $context->shop->id;
+        $query = pSQL(Tools::getValue('product_search'));
+
+        $sql = new DbQuery();
+        $sql->select('p.`id_product`, pa.`id_product_attribute`, pl.`name`, p.`reference`');
+        $sql->from('product', 'p');
+        $sql->join(Shop::addSqlAssociation('product', 'p'));
+        $sql->leftJoin('product_lang', 'pl', '
+            p.`id_product` = pl.`id_product`
+            AND pl.`id_lang` = ' . (int) $id_lang . Shop::addSqlRestrictionOnLang('pl')
+        );
+        $sql->leftJoin('product_attribute', 'pa', 'pa.`id_product` = p.`id_product`');
+        $where = 'pl.`name` LIKE \'%' . pSQL($query) . '%\'
+        OR p.`reference` LIKE \'%' . pSQL($query) . '%\'
+        OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_supplier` sp WHERE sp.`id_product` = p.`id_product` AND `product_supplier_reference` LIKE \'%' . pSQL($query) . '%\')';
+        $sql->orderBy('pl.`name` ASC');
+        if (Combination::isFeatureActive()) {
+            $where .= ' OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_attribute` `pa` WHERE pa.`id_product` = p.`id_product` AND (pa.`reference` LIKE \'%' . pSQL($query) . '%\'))';
+        }
+        $sql->where($where);
+        $result = Db::getInstance()->executeS($sql);
+        if (!$result) {
+            return false;
+        }
+        $results_array = array();
```

Comment:
You do not need to name it `$results_array` : it is obvious since we can read just after the variable name `= array();`

By the way, do you not use the `[]` notation on purpose for php backwards compatibility ?

Reply:
I do not use the [] because of the validator that returns an error.